contrib/profiler/profiler.py

```diff
@@ -0,0 +1,350 @@
+           sum(cpu_when_gpu_low) / length_gpu_low,
+           sum(mem_when_gpu_low) / length_gpu_low,
+           sum(disk_read_when_gpu_low) / length_gpu_low))
+    return [max_gpu, sum_gpu / count, max_cpu, sum_cpu / count, max_mem, sum_mem / count, max_read, sum_read,
+            sum(cpu_when_gpu_low) / length_gpu_low,
+            sum(mem_when_gpu_low) / length_gpu_low,
+            sum(disk_read_when_gpu_low) / length_gpu_low]
+
+
+def start_sample(container_id, period, analyze_period, duration, output_dir, gpu_id, container_pid):
+    start_time = time.time()
+    if not os.path.exists('./' + output_dir):
+        os.mkdir(output_dir)
+    realtime_log = csv.writer(open('./' + output_dir + '/log_result.csv', 'w'))  # , newline=''))
+    analyze_log = csv.writer(open('./' + output_dir + '/analyze_result.csv', 'w'))  # , newline=''))
+
+    str_write_realtime = ['cpu_usage', 'mem_used', 'IO_read', 'IO_write', 'network_receive', 'network_transmit']
+    for i in range(len(gpu_id)):
+        str_write_realtime.append('gpu_usage_' + str(gpu_id[i]))
+        str_write_realtime.append('gpu_mem_usage_' + str(gpu_id[i]))
+        str_write_realtime.append('gpu_mem_used_' + str(gpu_id[i]))
+        str_write_realtime.append('gpu_mem_total_' + str(gpu_id[i]))
+    realtime_log.writerow(str_write_realtime)
+
+    analyze_log.writerow(['max_gpu', 'avg_gpu', 'max_cpu', 'avg_cpu', 'max_mem', 'avg_mem', 'max_read', 'sum_read',
+                          'avg_cpu_gpu_low', 'avg_mem_gpu_low', 'avg_IO_gpu_low'])
+    nv.nvmlInit()
+    sample_list = list()
+    container_cpu_file_list = list()
+    container_mem_file_list = list()
+    container_blk_file_list = list()
+
+    print(
+        'max_gpu\tavg_gpu\tmax_cpu\tavg_cpu\tmax_memory\tavg_memory\tmax_read\ttotal_read\tavg_cpu_when_gpu_low'
+        '\tavg_mem_when_gpu_low\tavg_io_when_gpu_low')
+
+    if int(container_pid) == -1:
+        container_cpu_file_list.append('/sys/fs/cgroup/cpuacct/cpuacct.stat')
+        container_mem_file_list.append('/sys/fs/cgroup/memory/memory.usage_in_bytes')
+        container_blk_file_list.append('/sys/fs/cgroup/blkio/blkio.throttle.io_service_bytes')
+        container_net_file = '/proc/net/dev'
+    else:
+        container_cpu_file_list = glob.glob('/sys/fs/cgroup/cpuacct/docker/' + str(container_id) + '*/cpuacct.stat')
+        container_mem_file_list = glob.glob(
+            '/sys/fs/cgroup/memory/docker/' + str(container_id) + '*/memory.usage_in_bytes')
+        container_blk_file_list = glob.glob(
+            '/sys/fs/cgroup/blkio/docker/' + str(container_id) + '*/blkio.throttle.io_service_bytes')
+        container_net_file = '/proc/' + str(container_pid) + '/net/dev'
+    while time.time() - start_time < duration * 60:
+        [mem_used, mem_total] = get_memory_percent(container_mem_file_list)
+
+        # 1st info about I/O, network and CPU
+        read_bytes1 = get_disk_read_bytes(container_blk_file_list)
```

Comment:
Maybe you can wrap:
``` python
        read_bytes2 = get_disk_read_bytes(container_blk_file_list)
        write_bytes2 = get_disk_write_bytes(container_blk_file_list)
        [network_receive2, network_transmit2] = get_network_bytes(container_net_file)
        [sys_ticks2, container_ticks2] = get_cpu_ticks(container_cpu_file_list)
```
into one function. Since it called twice